scripts/basic-workflow-tests.sh

```diff
@@ -186,6 +186,22 @@ test_train() {
     fi
 
     ilab model train "${TRAIN_ARGS[@]}"
+
+    if [ "$FULLTRAIN" -eq 1 ]; then
+
+        # TODO Only cuda for now
+        ilab config init --non-interactive --train-profile="${SCRIPTDIR}/test-data/train-profile-a10.yaml"
+        TRAIN_ARGS=("--device=cuda" "--model-path=instructlab/granite-7b-lab" "--data-path=${SCRIPTDIR}/test-data/train_all_pruned_SDG.jsonl" "--quantize-data-type=nf4" "--4-bit-quant")
+        if [ "$GRANITE" -eq 1 ]; then
+            TRAIN_ARGS+=("--gguf-model-path models/granite-7b-lab-Q4_K_M.gguf")
+        fi
```

Comment:
Elsewhere in the script we conditionally put `--num-epochs 1` in the args? This code overwrites it. Is that option valid here? If so, can you replicate that logic so it's put back in place? The purpose is to speed up CI.

Reply:
the `train_profile` that @Maxusmusti and I put together takes care of the epochs I believe.

Reply:
yeah here it is: 
```
torch_args:
  nnodes: 1
  node_rank: 0
  nproc_per_node: 1
  rdzv_endpoint: 127.0.0.1:12222
  rdzv_id: 123
train_args:
  ckpt_output_dir: checkpoints
  data_output_dir: train-output
  data_path: ./taxonomy_data
  deepspeed_options:
    cpu_offload_optimizer: true
    cpu_offload_optimizer_pin_memory: false
    cpu_offload_optimizer_ratio: 1
    save_samples: null
  effective_batch_size: 100
  is_padding_free: false
  learning_rate: 2e-6
  lora:
    alpha: 32
    dropout: 0.1
    quantize_data_type: nf4
    rank: 2
    target_modules:
    - q_proj
    - k_proj
    - v_proj
    - o_proj
  max_batch_len: 1000
  max_seq_len: 96
  mock_data: false
  mock_data_len: 0
  model_path: instructlab/granite-7b-lab
  num_epochs: 1
  random_seed: 42
  save_samples: 100
  warmup_steps: 10
```

Reply:
OK, thanks. A comment about that would help. Someone may notice later "wait, we're overwriting TRAIN_ARGS here" -- maybe note that's intentional?